flink-runtime/src/main/java/org/apache/flink/runtime/deployment/ResultPartitionDeploymentDescriptor.java

```diff
@@ -58,6 +72,7 @@ public ResultPartitionDeploymentDescriptor(
 		KeyGroupRangeAssignment.checkParallelismPreconditions(maxParallelism);
 		this.maxParallelism = maxParallelism;
 		this.sendScheduleOrUpdateConsumersMessage = sendScheduleOrUpdateConsumersMessage;
+		this.isManagedExternally = partitionDescriptor.getPartitionType() == ResultPartitionType.BLOCKING;
```

Comment:
Since this property could be got directly based on `ResultPartitionType`, then another option is not defining this property explicitly in `ResultPartitionDeploymentDescriptor` and `ResultPartition` could also provide this property based on internal `ResultPartitionType`.

Reply:
So, my thinking is that the ResultPartitionType has no bearing on whether a partition can be consumed multiple times or is externally managed. These are all implementation details of the ShuffleEnvironment.
The only thing that the type itself specifies is whether or not it is consumable while it is being produced.

Now, so far we _did_ couple these things very explicitly to the type being BLOCKING (hence why this field is also based on the type), but long-term this might be problematic.
For example, we may at some point want to make pipelined partitions re-consumable as well, by also spilling them to disk while streaming them to producers at the same time.
If we bake this property into the type we'd have to add additional ResultPartitionTypes in the future to allow this to be done, which typically is a tricky think to do.

Additionally, the scheduler (at a later point) may decide that a certain blocking partition should not be managed by the JM but by the environment, which a separate flag would allow.
(At which point we'D have to modify the constructor obviously)

Reply:
Agree, thanks for the explanation!

Reply:
Agree, thanks for the explanation!